Nullable integers and string dtype

vadella opened the pull request:
Add support for the new dtypes in `utils.sanitize_dataframe` to solve #1922

- [nullable integer datatype](https://pandas.pydata.org/pandas-docs/version/0.25/whatsnew/v0.24.0.html#optional-integer-na-support) 
- [dedicated string dtype](https://pandas.pydata.org/pandas-docs/version/1.0.0/whatsnew/v1.0.0.html#dedicated-string-data-type)

vadella commented:
To make the tests pass for pandas versions which do not support the new dtypes, I surrounded it with 
```
try:
    ...
except TypeError:  # dtype not supported
    return
```

An alternative here would be 
```
try:
    ...
except TypeError:  # dtype not supported
    assert pd.__version__ <= <min_pandas_version_for_feature>
    return
```

Would that be better?

jakevdp commented on altair/utils/tests/test_utils.py:
```diff
@@ -106,3 +106,61 @@ def test_sanitize_dataframe_infs():
     assert list(df_clean['x']) == [0, 1, 2, None, None, None]
+
+
+def test_sanitize_nullable_integers():
```
Rather than silently passing if there is a type error, it would be better to explicitly skip this test for older versions of pandas. The cleanest way would be to add a decorator like:
```python
@pytest.mark.skipif(not hasattr(pd, "Int64Dtype"), "Nullable integers not supported in pandas v{}".format(pd.__version__))
def test_sanitize_nullable_integers():
    ...
```
and then remove the ``try...except`` statement.

Similar for the string test below.

vadella commented:
I've changed it to use `pytest.mark.skipif`

I've also added the behaviour to the docstring.

Initially I forgot the nullable boolean dtype. 
For clarity, I added this as a separate path in the `elif` tree, instead of adding it to  one of the other branches even though what happens is the same as for example the `category` dtype.

jakevdp commented:
Looks great, thanks!
